registry/registry_db.go

```diff
@@ -40,20 +40,50 @@ import (
 	"github.com/pelicanplatform/pelican/utils"
 )
 
+type RegistrationStatus int
+
+// The AdminMetadata is used in [Namespace] as a marshalled JSON string
+// to be stored in registry DB.
+//
+// The *UserID are meant to correspond to the "sub" claim of the user token that
+// the OAuth client issues if the user is logged in using OAuth, or it should be
+// "admin" from local password-based authentication.
+//
+// To prevent users from writing to certain fields (readonly), you may use "post" tag
+// with value "exclude". This will exclude the field from user's create/update requests
+// and the field will also be excluded from field discovery endpoint (OPTION method).
+//
+// We use validator package to validate struct fields from user requests. If a field is
+// required, add `validate:"required"` to that field. This tag will also be used by fields discovery
+// endpoint to tell the UI if a field is required. For other validator tags,
+// visit: https://pkg.go.dev/github.com/go-playground/validator/v10
+type AdminMetadata struct {
+	UserID                string             `json:"user_id" post:"exclude"` // "sub" claim of user JWT who requested registration
+	Description           string             `json:"description"`
+	SiteName              string             `json:"site_name" validate:"required"`
```

Comment:
Let's make the SiteName field not-required.

In OSG Topology, the "site name" is an organization within an institution (topology calls this a "facility").  For example, a university may have a "Physics Department" and "CS Department" that run independent resources.  However, the site name is only used to group contacts for services (a shortcut for putting the same person as in charge of every service at the site).

In Pelican, I'd like to see how far we can push things without having a "site name" concept.  We may come back to it -- but I'd like to start off with it being optional.